include/swift/RemoteInspection/ExternalDescriptorFinder.h

```diff
@@ -0,0 +1,56 @@
+//===--------------- ExternalDescriptorFinder.h -----------------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_REFLECTION_EXTERNAL_DESCRIPTOR_FINDER_H
+#define SWIFT_REFLECTION_EXTERNAL_DESCRIPTOR_FINDER_H
+
+#include "llvm/ADT/StringRef.h"
+
+namespace swift {
+namespace reflection {
+
+class TypeRef;
+
+enum class FieldDescriptorKind : uint16_t;
+
+/// A generic interface a builtin type descriptor.
+struct BuiltinTypeDescriptorInterface {
+  const uint32_t Size;
+  const uint32_t Alignment;
+  const uint32_t Stride;
+  const uint32_t NumExtraInhabitants;
+  const bool IsBitwiseTakable;
+
+  BuiltinTypeDescriptorInterface(uint32_t Size, uint32_t Alignment,
+                                 uint32_t Stride, uint32_t NumExtraInhabitants,
+                                 bool IsBitwiseTakable)
+      : Size(Size), Alignment(Alignment), Stride(Stride),
+        NumExtraInhabitants(NumExtraInhabitants),
+        IsBitwiseTakable(IsBitwiseTakable) {}
+
+  virtual ~BuiltinTypeDescriptorInterface() {};
+
+  virtual llvm::StringRef readMangledTypeName() = 0;
```

Comment:
`getMangledTypeName()`?
the "read" part is only true for the reflection metadata implementation, right?